modules/src/twitter.py

```diff
@@ -0,0 +1,130 @@
+import json
+import requests
+import config
+import os
+from requests.utils import quote
+from requests_oauthlib import OAuth1
+from templates.generic import *
+from templates.text import TextTemplate
+
+TWI_CONSUMER_KEY = os.environ.get('TWI_CONSUMER_KEY', config.TWI_CONSUMER_KEY)
+TWI_CONSUMER_SECRET = os.environ.get(
+    'TWI_CONSUMER_SECRET', config.TWI_CONSUMER_SECRET)
+TWI_ACCESS_TOKEN = os.environ.get('TWI_ACCESS_TOKEN', config.TWI_ACCESS_TOKEN)
+TWI_ACCESS_TOKEN_SECRET = os.environ.get(
+    'TWI_ACCESS_TOKEN_SECRET', config.TWI_ACCESS_TOKEN_SECRET)
+
+TWI_URL_SEARCH_API = 'https://api.twitter.com/1.1/search/tweets.json?q='
+TWI_USER_TIMELINE_API = 'https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name='
+
+params = {'search':
+          {'count': 1,
+           'result_type': 'mixed',
+           'include_entities': 'true',
+           'extended_tweet': 'true'
+           },
+          'timeline':
+          {'count': 1,
+           'exclude_replies': 'true',
+           'extended_tweet': 'true'
+           }
+          }
+
+
+def get_oauth():
+    """ Generates an OAuth object """
+
+    oauth = OAuth1(TWI_CONSUMER_KEY,
+                   client_secret=TWI_CONSUMER_SECRET,
+                   resource_owner_key=TWI_ACCESS_TOKEN,
+                   resource_owner_secret=TWI_ACCESS_TOKEN_SECRET)
+    return oauth
+
+
+def getUrl(tweet):
+    """ getUrl by author or popular topic """
+
+    if tweet[0] == '@':
+        author = tweet[1:]
+        req = requests.get(url=TWI_USER_TIMELINE_API + author, headers={
+                           'content-type': 'application/json'}, params=params['timeline'], auth=get_oauth())
+        req = req.json()[0]
```

Comment:
This might fail if response fails, let's check the response first :), it would be nice to name the variable response/resp so that it is relevant. :)

Reply:
This is catch by try/except lines 113 to 127.
Any non-managed exception on the tweet process are handle this way.
Could be relevant to try to catch earlier but only for log purposes, can't deliver an HTTP error processed by the server to the client side.

Reply:
Hey @ZuZuD  thanks for pointing out, overlooked it, but yes. Quite not happy about how the exceptions are caught in the line you specified, that is very vague. However, all the modules are written like you have done, so no comments there. Good to go from my side, unless @swapagarwal  has something to add 👍